src/components/main/Canvas.js

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { makeStyles } from '@material-ui/core/styles';
+import ObjectID from 'bson-objectid';
+import Form from './form/Form';
+import Note from './note/Note';
+import {
+  setActiveForm,
+  addNote,
+  postAppInstanceResource,
+  getAppInstanceResources,
+} from '../../actions';
+import { generateRandomRotationAngle } from '../../utils/canvas';
+import Settings from '../modes/teacher/Settings';
+import {
+  NOTE_STRING,
+  IMAGE_STRING,
+  RE_FETCH_INTERVAL,
+} from '../../constants/constants';
+import { TEACHER_MODE } from '../../config/settings';
+
+const useStyles = makeStyles(() => ({
+  mainContainer: {
+    width: '100%',
+    height: '100%',
+    cursor: 'cell',
+    background: '#FFE4E1',
+  },
+  image: { width: '100%', height: '100%', display: 'block' },
+}));
+
+const Canvas = () => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+
+  // extract required state from redux store
+  const { mode, standalone } = useSelector(({ context }) => context);
+  const { activeForm, notes: sessionNotes } = useSelector(
+    ({ canvas }) => canvas,
+  );
+  // in this app, appInstanceResources can be of different types (note or background image)
+  // hence, filter retrieved appInstanceResources by type
+  const savedNotes = useSelector(({ appInstanceResources }) =>
+    appInstanceResources.content.filter(
+      (content) => content.data.type === NOTE_STRING,
+    ),
+  );
+  const backgroundImages = useSelector(({ appInstanceResources }) =>
+    appInstanceResources.content.filter(
+      (content) => content.data.type === IMAGE_STRING,
+    ),
+  );
+  const userId = useSelector(({ context }) => context.userId);
+
+  // sort retrieved background images by date, and then take the most recent one
+  const mostRecentImage = backgroundImages.sort(
+    (imageA, imageB) => imageA.data.date > imageB.data.date,
+  )[0];
```

Comment:
Interesting. I thought the ids in the DB were stored as strings (mongo IDs)? I did not realize these strings are ordered. In any case, since we are now storing the background image in appInstance settings, this approach no longer applies.